src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,168 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elementData = new Object[]{};
+
+    }
+
+    public ArrayList(int customCapacity) {
+        if (customCapacity > 0) {
+            this.elementData = new Object[customCapacity];
+        } else if (customCapacity == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+        } else {
+            throw new RuntimeException("Impossible capacity: "
+                    + customCapacity);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow(size + 1);
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        if (size == elementData.length) {
```

Comment:
Similar to the previous comment, the `add` method with index parameter should not directly handle the array resizing. Extract the growth logic into a separate method.